### PR TITLE
exclude `functional_test_junit_upload_system_probe_kmt_*` from merge queue

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

`kernel_matrix_testing_run_tests_*` jobs are not included in the merge queue, so no need for the jobs upload CI test results to datadog.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/junit_upload/functional_test_junit_upload_system_probe.yml [hunk 1]
```diff
@@ -29,6 +29,8 @@ functional_test_junit_upload_system_probe:
   tags: ["arch:amd64"]
   allow_failure: true
   when: always
+  rules:
+    - !reference [.except_mergequeue]
```
Comment (chouetz) [inline .gitlab/junit_upload/functional_test_junit_upload_system_probe.yml:33]:
```suggestion
    - !reference [.except_mergequeue]
    - when: on_success
```
I think we need this (at least to be consistent with the other place it's defined)

**Review by chouetz — CHANGES_REQUESTED** [review]

File: .gitlab/junit_upload/functional_test_junit_upload_system_probe.yml (hunk 1 above)
Comment (paulcacheux) [reply .gitlab/junit_upload/functional_test_junit_upload_system_probe.yml:32]: Won't this mean that the job doesn't run if previous stages failed ? i.e. we would only upload when everything is green ?

Comment (spencergilbert) [reply .gitlab/junit_upload/functional_test_junit_upload_system_probe.yml:32]: 🤔 I hate `rules`, that seems correct reading it here - if that's the case we'll want to update that elsewhere.

Comment (paulcacheux) [reply .gitlab/junit_upload/functional_test_junit_upload_system_probe.yml:31]: I updated the other functional tests upload jobs, I think this makes it consistent

The regular upload jobs don't have this issue AFAIU

**Review by chouetz — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**